10week/checkpoint3/test.js

```diff
@@ -0,0 +1,379 @@
+'use strict'
+
+class CurrentEQ extends React.Component {
+  constructor(props) {
+    super(props);
+
+    this.state = {
+      results: [],
+      quakes: [],
+      filter: [],
+      lessThan: [],
+      greaterThan: [],
+      total: null,
+      average: null,
+      stdDev: null,
+      bins: [0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0]
+    }
+    this.monthGather = this.monthGather.bind(this);
+    this.weekGather = this.weekGather.bind(this);
+    this.dayGather = this.dayGather.bind(this);
+    this.clearData = this.clearData.bind(this);
+    this.magGather = this.magGather.bind(this);
+    this.average = this.average.bind(this);
+    this.standardDeviation = this.standardDeviation.bind(this);
+    this.lessThan = this.lessThan.bind(this);
+    this.greaterThan = this.greaterThan.bind(this);
+    this.bins = this.bins.bind(this)
+  }
+
+  // for filter test:
+  // lessThan(num){
+  //   return
+  // }
+
+
+  lessThan(){
+    // Get the value of the input field with id="filter1"
+    let y = document.getElementById("filter1").value;
+    let x = Number(y);
+    // If x is Not a Number or less than one or greater than 10
+    if (isNaN(x) || x <= 4.5 || x > 10) {
+      alert("Input not valid");
+    } else {
+      let newList1 = [];
+      let filtList1 = this.state.filter;
+      let list1 = this.state.quakes;
+      for (var i=0; i<list1.length;i++){
+        let equake = list1[i];
+        if (equake >= x) {
+          newList1.push(equake);
+        } else {
+          filtList1.push(equake);
+        }
+        // console.log('newList1.length:',newList1.length, 'filtList1.length:', filtList1.length);
+        let newSTD = this.standardDeviation(newList1);
+        let newBin = this.bins(newList1);
+        this.setState({
+          quakes: newList1 ,
+          lessThan: newList1 ,
+          filter: filtList1 ,
+          total: newList1.length,
+        });
+      }
+    }
+  }
+
+  greaterThan(){
+    // Get the value of the input field with id="filter1"
+    let y = document.getElementById("filter2").value;
+    let x = Number(y);
+    // If x is Not a Number or less than one or greater than 10
+    if (isNaN(x) || x <= 4.5 || x > 10) {
+      alert("Input not valid");
+    } else {
+      let newList2 = [];
+      let filtList2 = this.state.filter;
+      let list2 = this.state.quakes;
+      for (var i=0; i<list2.length;i++){
+        let equake = list2[i];
+        if (equake <= x) {
+          newList2.push(equake);
+        } else {
+          filtList2.push(equake);
+        }
+        // console.log('newList2.length:',newList2.length, 'filtList2.length:', filtList2.length);
+        let newSTD = this.standardDeviation(newList2);
+        let newBin = this.bins(newList2);
+        this.setState({
+          quakes: newList2 ,
+          lessThan: newList2 ,
+          filter: filtList2 ,
+          total: newList2.length,
+        });
+      }
+    }
+  }
+  bins(data){
```

Comment:
Instead of having an array of bins, where the first element is 4.5, second is 4.6  and so on, You could have used a map where the key is "45", "46", "47", and so on.
That would have reduced the need to have so many hardcoded parts.